src/main/java/stsjorbsmod/patches/CodaPatch.java

```diff
@@ -0,0 +1,36 @@
+package stsjorbsmod.patches;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpireField;
+import com.evacipated.cardcrawl.modthespire.lib.SpireInsertPatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import javassist.CannotCompileException;
+import javassist.expr.ExprEditor;
+import javassist.expr.MethodCall;
+
+@SpirePatch(clz = DamageInfo.class, method = SpirePatch.CLASS)
+public class CodaPatch {
+    public static SpireField<Integer> hpRetained = new SpireField<>(() -> 0);
+
+    public static void retainHP(AbstractCreature __this, DamageInfo info) {
+        if (__this.currentHealth <= 0 && hpRetained.get(info) > 0) {
+            __this.currentHealth = hpRetained.get(info);
+        }
+    }
+
+    @SpirePatch(clz = AbstractMonster.class, method = "damage")
+    public static class AbstractMonster_damage {
+        public static ExprEditor Instrument() {
+            return new ExprEditor(){
+                @Override
+                public void edit(MethodCall methodCall) throws CannotCompileException {
+                    if (methodCall.getClassName().equals(AbstractMonster.class.getName()) && methodCall.getMethodName().equals("healthBarUpdatedEvent")) {
```

Comment:
I think it would probably be a little friendlier to other effects to leave `this.lastDamageTaken` (set higher up than `healthBarUpdatedEvent` in `AbstractMonster::damage`) accurate to what Coda ends up dealing; I'd recommend patching in right before the field access to `lastDamageTaken` instead to check for hpRetained and clamp to `this.currentHealth - hpRetained.get(info)`